examples/high_frequency/emc/double_pulse_test.py

```diff
@@ -0,0 +1,382 @@
+    "l_load": "7.7e-5",
+    "Vgate_top": "-5V",
+    "r_load": "0.1",
+    "r_dc_link": "1e6",
+    "v_pwl_high": "18.0V",
+    "v_pwl_low": "-5.0V",
+}
+
+# Define design variables from the created dictionaries.
+for k, v in des_properties.items():
+    circuit[k] = v
+
+# ## Insert Circuit Elements into the Schematic
+#
+y_upper_pin = 5200
+y_lower_pin = 2000
+time_list_pwl = [
+    0.0,
+    5.0e-6,
+    5.001e-6,
+    9.825e-06,
+    9.826e-06,
+    1.1e-05,
+    1.1001e-05,
+    1.3e-05,
+    1.3001e-05,
+]
+volt_list_pwl = [
+    "v_pwl_low",
+    "v_pwl_low",
+    "v_pwl_high",
+    "v_pwl_high",
+    "v_pwl_low",
+    "v_pwl_low",
+    "v_pwl_high",
+    "v_pwl_high",
+    "v_pwl_low",
+]
+
+# Add circuit components to the schematic.
+v_pwl = circuit.modeler.components.create_voltage_pwl(
+    name="v_pwl",
+    time_list=time_list_pwl,
+    voltage_list=volt_list_pwl,
+    location=[600, 2800],
+)
+v_gate_top = circuit.modeler.components.create_voltage_dc(
+    name="Vgate_top", value="Vgate_top", location=[600, 4500]
+)
+v_dc_bus = circuit.modeler.components.create_voltage_dc(
+    name="v_dc_bus", value="VoltageDCbus", location=[-1800, 3800]
+)
+c_dc_link = circuit.modeler.schematic.create_capacitor(
+    name="c_dc_link", value="c_dc_link", location=[-1300, 3800], angle=90
+)
+r_dc_link = circuit.modeler.schematic.create_resistor(
+    name="r_dc_link", value="r_dc_link", location=[-700, 3800], angle=90
+)
+l_load = circuit.modeler.schematic.create_inductor(
+    name="l_load", value="l_load", location=[3000, 4800], angle=-90
+)
+r_load = circuit.modeler.schematic.create_resistor(
+    name="r_load", value="r_load", location=[3000, 4300], angle=90
+)
+r_g1 = circuit.modeler.schematic.create_resistor(
+    name="r_g1", value="r_g1", location=[1400, 4700], angle=180
+)
+r_g2 = circuit.modeler.schematic.create_resistor(
+    name="r_g2", value="r_g2", location=[1400, 3100], angle=180
+)
+voltm_g = circuit.modeler.components.components_catalog["Probes:VPROBE_DIFF"].place(
+    assignment="voltage_g", location=[100, 2900], angle=270
+)
+voltm_g.parameters["Name"] = "voltage_g"
+voltm_ds = circuit.modeler.components.components_catalog["Probes:VPROBE_DIFF"].place(
+    assignment="voltage_ds", location=[2500, 3300], angle=0
+)
+voltm_ds.parameters["Name"] = "voltage_ds"
+amm_top = circuit.modeler.components.components_catalog["Probes:IPROBE"].place(
+    assignment="Itop", location=[1100, 5200], angle=0
+)
+amm_top.parameters["Name"] = "Itop"
+amm_ind = circuit.modeler.components.components_catalog["Probes:IPROBE"].place(
+    assignment="Iinductor", location=[2500, 4000], angle=0
+)
+amm_ind.parameters["Name"] = "Iinductor"
+amm_bot = circuit.modeler.components.components_catalog["Probes:IPROBE"].place(
+    assignment="Ibottom", location=[2000, 3600], angle=270
+)
+amm_bot.parameters["Name"] = "Ibottom"
+
+
+# ## Add nmos components from Component Library.
+#
+# Please check that chosen component has the attribute .place
```

Comment:
```suggestion

```
What do you mean? components_catalog[] returns a ComponentInfo object and you can access the place method.

Reply:
not all the Circuit component have it (i tried some are lacking)